Sources/System/FilePath/FilePathSyntax.swift

```diff
@@ -0,0 +1,353 @@
+// Query API
+// @available(...)
+extension FilePath {
+  /// Returns true if this path uniquely identifies the location of
+  /// a file without reference to an additional starting location.
+  ///
+  /// On Unix platforms, absolute paths begin with a `/`. `isAbsolute` is equivalent
+  /// to `root != nil`.
+  ///
+  /// On Windows, absolute paths are fully qualified paths. UNC paths and device paths
+  /// are always absolute. Traditional DOS paths are absolute if they begin with a volume or drive
+  /// followed by a `:` and a separator.
+  ///
+  /// NOTE: This does not perform shell expansion or substitute
+  /// environment variables; paths beginning with `~` are considered relative.
+  ///
+  /// Examples:
+  /// * Unix:
+  ///   * `/usr/local/bin`
+  ///   * `/tmp/foo.txt`
+  ///   * `/`
+  /// * Windows:
+  ///   * `C:\Users\`
+  ///   * `\\?\UNC\server\share\bar.exe`
+  ///   * `\\server\share\bar.exe`
+  ///
+  /// TODO: We should add API to system to access environment variables
+  /// and perform expansion/substituion, but that's deferred for now. Should
+  /// we mention that?
+  public var isAbsolute: Bool {
```

Comment:
I still think that `RelativeFilePath` and `AbsoluteFilePath` (possibly shortened to just `FilePath`) should be primary types, so that API can declare their assumptions at compile-time rather than throw runtime errors if the "wrong" kind of path is passed.  I understand that this isn't how most APIs represent path types, but I think it's worked quite well in ToolSupportCore and SwiftPM.

Reply:
We could consider adding a `FilePath` protocol and concrete types that conform and precondition on construction. This can be added at any time and I'd prefer to defer that until after we add semantic operations, since I think a lot of the relative-vs-absolute concerns come up then.

IIRC, SPM doesn't even use the same semantics for absolute paths as we do here, so you might not get what you want if System provides these types. This would argue to defer this until we have access to the environment for `~` expansion, and even `realpath` functionality.

Reply:
Understood.  I do think that `~` expansion and `realpath` would probably be outside of a `FilePath` value type (especially `realpath`, since it depends on which file system is involved), but that would be a later discussion as well.